Dockerfile

```diff
@@ -96,15 +104,42 @@ ARG GIT_COMMIT_ID
 RUN PKG_CONFIG="/usr/bin/$(xx-info)-pkg-config" \
     PATH="$PATH:/opt/mold/bin" \
     RUSTFLAGS="${LINKER:+-C link-arg=-fuse-ld=}$LINKER $RUSTFLAGS" \
-    xx-cargo build --profile $PROFILE ${FEATURES:+--features} $FEATURES --features=stacktrace --bin qdrant \
+    xx-cargo build --profile $PROFILE ${FEATURES:+--features} $FEATURES --features=stacktrace ${GPU:+--features=gpu} --bin qdrant \
     && PROFILE_DIR=$(if [ "$PROFILE" = dev ]; then echo debug; else echo $PROFILE; fi) \
     && mv target/$(xx-cargo --print-target-triple)/$PROFILE_DIR/qdrant /qdrant/qdrant
 
 # Download and extract web UI
 RUN mkdir /static && STATIC_DIR=/static ./tools/sync-web-ui.sh
 
 
-FROM debian:12-slim AS qdrant
+# Dockerfile does not support conditional `FROM` directly.
+# To workaround this limitation, we use a multi-stage build with a different base images which have equal name to ARG value.
+
+# Base image for Qdrant.
+FROM debian:12-slim AS qdrant-cpu
+
+
+# Base images for Qdrant with nvidia GPU support.
+FROM nvidia/opengl:1.0-glvnd-devel-ubuntu22.04 AS qdrant-gpu-nvidia
+# Set non-interactive mode for apt-get.
+ARG DEBIAN_FRONTEND=noninteractive
```

Comment:
Also, is this necessary? And if so, we should be able to move it to the top of `AS qdrant` step? And change this from `ARG DEBIAN_FRONTEND` to `ENV DEBIAN_FRONTEND`?

Reply:
Figured why `ARG` is preferable here, but I'd still move it to the top of `AS qdrant` step.

Reply:
I checked that `ENV` works fine, so I changed it to env, thanks! I don't want to affect the regular build so I prefer to keep param out of `AS qdrant` step.